libretro-common/include/libretro.h

```diff
@@ -976,6 +977,85 @@ struct retro_hw_render_context_negotiation_interface
+  bool isDirectory;          /* The stat url is a directory */
+  bool isSymLink;            /* The stat url is a symbolic link */
+  bool isHidden;             /* The file is hidden */
+};
+
+/* Opaque file handle */
+typedef struct retro_file retro_file;
+
+/* File operations */
+typedef retro_file* (RETRO_CALLCONV *retro_open_file_t)(const char *url, int flags);
+typedef retro_file* (RETRO_CALLCONV *retro_open_file_for_write_t)(const char *url, bool overwrite);
+typedef int64_t (RETRO_CALLCONV *retro_read_file_t)(retro_file *file, uint8_t *buffer, size_t buffer_size);
+typedef int64_t (RETRO_CALLCONV *retro_write_file_t)(retro_file *file, const uint8_t *buffer, size_t buffer_size);
+typedef int64_t (RETRO_CALLCONV *retro_seek_file_t)(retro_file *file, int64_t position, retro_seek_flag whence);
+typedef int64_t (RETRO_CALLCONV *retro_get_file_position_t)(retro_file *file);
+typedef int64_t (RETRO_CALLCONV *retro_get_file_length_t)(retro_file *file);
+typedef int64_t (RETRO_CALLCONV *retro_close_file_t)(retro_file *file);
+typedef int (RETRO_CALLCONV *retro_get_file_chunk_size_t)(retro_file *file);
+typedef bool (RETRO_CALLCONV *retro_file_exists_t)(const char *url);
+typedef bool (RETRO_CALLCONV *retro_stat_file_t)(const char *url, struct retro_stat_struct *buffer);
+
+struct retro_dir_entry
+{
+  char *label;   /* Item label */
+  char *path;    /* Item path */
+  bool folder;   /* Item is a folder */
+  uint64_t size; /* Size of file represented by item */
+};
+
+/* Directory operations */
+typedef bool (RETRO_CALLCONV *retro_can_open_directory_t)(const char *url);
+typedef bool (RETRO_CALLCONV *retro_create_directory_t)(const char *url);
+typedef bool (RETRO_CALLCONV *retro_directory_exists_t)(const char *url);
+typedef bool (RETRO_CALLCONV *retro_remove_directory_t)(const char *url);
+typedef bool (RETRO_CALLCONV *retro_get_directory_t)(const char *url, const char *mask, retro_dir_entry **items, unsigned int *item_count);
+typedef void (RETRO_CALLCONV *retro_free_directory_t)(retro_dir_entry *items, unsigned int item_count);
+
+struct retro_vfs_interface
+{
+  retro_open_file_t retro_open_file;
+  retro_open_file_for_write_t retro_open_file_for_write;
+  retro_read_file_t retro_read_file;
+  retro_write_file_t retro_write_file;
+  retro_seek_file_t retro_seek_file;
+  retro_get_file_position_t retro_get_file_position;
+  retro_get_file_length_t retro_get_file_length;
+  retro_close_file_t retro_close_file;
+  retro_get_file_chunk_size_t retro_get_file_chunk_size;
+  retro_file_exists_t retro_file_exists;
+  retro_stat_file_t retro_stat_file;
+  retro_can_open_directory_t retro_can_open_directory;
+  retro_create_directory_t retro_create_directory;
+  retro_directory_exists_t retro_directory_exists;
+  retro_remove_directory_t retro_remove_directory;
+  retro_get_directory_t retro_get_directory;
+  retro_free_directory_t retro_free_directory;
+};
+
+#define RETRO_ENVIRONMENT_GET_VFS_INTERFACE 45
```

Comment:
44 being the savestate quirks I guess? +1 for attention, but we'll get a merge conflict anyways.